### ISSUE TITLE
Discussion: React Native for Windows Roadmap and July updates

### ISSUE THREAD

**Issue by harinikmsft** [microsoft/react-native-windows#2815]
# Updates
We announced the first public preview of React Native for Windows - vnext project at Microsoft BUILD 2019 conference earlier this year. You can watch the presentation - [React Native: Build rich Windows and mobile applications easily using JavaScript](https://mybuild.techcommunity.microsoft.com/sessions/77007) for more details.

We have since been making steady progress in the past several months, you can read about these in our [July updates blog](https://github.com/microsoft/react-native-windows/blob/master/blog/July2019Updates.md). 

# Roadmap feedback
We recently published a high level [prospective roadmap](https://github.com/microsoft/react-native-windows/blob/master/blog/Roadmap2019.md) on React Native for Windows for 2019 through early 2020. We would like to hear your feedback and have some questions below:

-	What do you think about the overall plan outlined in the roadmap? Would it enable you to use React Native for Windows vNext for your new and existing apps?

-	Are there major high level features (or) updates that you are looking for that our roadmap does not cover towards the first stable RC later this year? 
_For example_ – we are working on adding __[Hermes engine support](https://github.com/facebook/hermes)__ to React Native Windows in the ongoing [Milestone 3](https://github.com/microsoft/react-native-windows/milestone/17) (see Issues [2799](https://github.com/microsoft/react-native-windows/issues/2799) and [58](https://github.com/facebook/hermes/issues/58)). 

      Are there other major features or updates you would like to see us tackling right away? Are there any specific tooling related investments you are looking for?

-	__Version updates__: We currently support React Native version 0.59. Our current plans are to try and keep up with React Native versions as we accomplish more feature parity. This would mean, we will be updating from 0.59 to 0.60 and then 0.61(?) as we work towards the stable release candidate later this year. This means, there is likely to be feature gaps between React-Native and React-Native-Windows vNext while targeting the same version. 

       We debated locking to an important version (say 0.59) and completing all the features needed for full parity – but landed on this approach instead as that helps us (and by extension, you) keep up with React Native versioning better. We would be willing to consider back-porting/fixing select features or issues in a previous version if enough customers are blocked by it. Do you have any feedback on this approach?

-	__Extensions__: We are working on adding APIs for native extensions, modules and view managers story for React Native for Windows vNext. Our intention is to first closely match the APIs from the [“current” version](https://github.com/microsoft/react-native-windows/blob/master/current/docs/NativeModulesWindows.md). We will be looking into [Turbo modules](https://github.com/react-native-community/discussions-and-proposals/issues/40), [Expo unimodules](https://docs.expo.io/versions/v33.0.0/bare/hello-world/#using-unimodulescore) in parallel. We would love to hear your feedback on this topic – what would you like to see us invest in this space?

**Comment by brunolemos** [comment]
I have this cross-platform project called [DevHub](https://github.com/devhubapp/devhub) which currently supports iOS/Android (react-native) and Web (react-native-web).

It’s on my mid-term plans to have a react-native-windows version as well, but that’s still a few months away.

I invite anyone here to clone the project and experiment adding react-native-windows to it. If it works well than I believe react-native-windows is on a good state to be released. 

Should be a good way to find unintended  platform differences as well.

**Comment by sami616** [comment]
This project looks fantastic, cant wait to get stuck and building desktop apps with react!

**Comment by ShrinathGupta** [comment]
Hi all,
There is ongoing parallel development of [demo](https://github.com/ShrinathGupta/React-native-windows-demo) of UWP application on react-native-windows vnext version.
I have used few UI components in demo. I also invite everyone to clone the repository and play with it. It might help you to set-up environment, understand code structure. 

Few UI components are still not stable like modal, drop down etc. I will be going to update all the stable components very soon.

**Comment by harinikmsft** [comment]
Thank you @ShrinathGupta. Love the app! We are tracking all the remaining issues for getting to parity with React Native through the [Parity Status Doc](https://github.com/microsoft/react-native-windows/blob/master/vnext/docs/ParityStatus.md). Do feel free to file issues for us if you notice any APIs not working as expected and if they are not already tracked by us.

We should have the native modules, view managers story ready shortly - that should help you to add even more components to your app!

**Comment by ShrinathGupta** [comment]
You're welcome @harinikmsft . Thanks for sharing status doc link. 
I would love to add more components to my app. 

All, I have also published an article that will help to understand more about uwp application. [Article Link](https://medium.com/@shrinath161/universal-windows-platform-application-by-using-react-native-windows-vnext-92d8c65b304b)
Thanks.!

**Comment by leoek** [comment]
@harinikmsft Is there any more specific eta for the RN 0.60 upgrade than summer-fall 2019? We are considering to split our codebase and upgrade Android/IOS to RN 0.60 while keeping Windows in a legacy branch. However I would much rather wait for a few weeks, if react-native-windows is close to 0.60 and upgrade all three to RN 0.60.

**Comment by steipete** [comment]
Looking at https://github.com/microsoft/react-native-windows/issues/2786 it seems they target end of October as 0.60 release.

**Comment by harinikmsft** [comment]
@leoek we completed the upgrade to RN 0.60 last week for `vnext` ([Issue 2786 closed](https://github.com/microsoft/react-native-windows/issues/2786)).

**Comment by leoek** [comment]
@harinikmsft Thanks! Unfortunately we do need it for `current` as I mentioned in this  https://github.com/PSPDFKit/react-native/issues/294#issuecomment-538083829

**Comment by nickwinder** [comment]
@harinikmsft Could I ask what the plan is for upgrading `current` to RN 0.60? It is still a requirement of https://github.com/PSPDFKit/react-native/issues/294 and would like to know how to address the issue.

**Comment by leoek** [comment]
@harinikmsft do you have an eta for 0.60 current for us?

**Comment by NickGerleman** [comment]
Closing since there hasn't been recent activity.
